strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/repository/NugetRepositoryFeatures.java

```diff
@@ -1,10 +1,112 @@
 package org.carlspring.strongbox.repository;
 
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.carlspring.strongbox.artifact.coordinates.NugetHierarchicalArtifactCoordinates;
+import org.carlspring.strongbox.client.ArtifactTransportException;
+import org.carlspring.strongbox.configuration.Configuration;
+import org.carlspring.strongbox.configuration.ConfigurationManager;
+import org.carlspring.strongbox.domain.RemoteArtifactEntry;
+import org.carlspring.strongbox.services.ArtifactEntryService;
+import org.carlspring.strongbox.storage.Storage;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.storage.repository.remote.RemoteRepository;
+import org.springframework.stereotype.Component;
+
+import ru.aristar.jnuget.client.NugetClient;
+import ru.aristar.jnuget.rss.PackageEntry;
+import ru.aristar.jnuget.rss.PackageFeed;
+
 /**
+ * 
  * @author carlspring
+ * @author Sergey Bespalov
  */
+@Component
 public class NugetRepositoryFeatures
         implements RepositoryFeatures
 {
 
+    @Inject
+    private ConfigurationManager configurationManager;
+
+    @Inject
+    private ArtifactEntryService artifactEntryService;
+
+    public void downloadRemoteIndex(String storageId,
```

Comment:
In this case, I assume a better name for this method, would probably be `downloadRemoteFeed`...? This is what I was referring to in my earlier comment regarding the method that you defined in the interface -- in one implementation, the "features" may be called `X`, `Y`, `Z`, while in another -- there could be `Y` and `A`. The `ReposiotryFeatures` interface defines no methods for this exact reason. However, there is a mechanism in place that uses these "features".

Reply:
ok, I will rename it
thanks